pkg/asset/installconfig/azure/validation.go

```diff
@@ -177,3 +178,53 @@ func ValidatePublicDNS(ic *types.InstallConfig, azureDNS *DNSConfig) error {
 
 	return nil
 }
+
+// ValidateForProvisioning validates if the isntall config if valid for provisioning the cluster.
+func ValidateForProvisioning(client API, ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateResourceGroup(client, field.NewPath("platform").Child("azure"), ic.Azure)...)
+	return allErrs.ToAggregate()
+}
+
+func validateResourceGroup(client API, fieldPath *field.Path, platform *aztypes.Platform) field.ErrorList {
```

Comment:
Need to validate if resourceGroupName provided complies with azure naming:
```
// https://docs.microsoft.com/en-us/azure/architecture/best-practices/resource-naming#general
var resourceGroupNameRx = regexp.MustCompile(`(?i)^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
if !resourceGroupNameRx.MatchString(ans.(string)) {
      return errors.Errorf("invalid resource group %q", ans.(string))
}
```

Reply:
It expects an existing resource group so not sure if we need this validation on the name itself

Reply:
It is good practice to always validate user inputs. The fact that validation was done by 3rd party (azure while creation RG) is not right. 
Hypothetical situation. In 6-month team members rotated, and somebody else comes in and implements "IfNotExistCreate" functionality. And you have a failure on your ends if they are not aware of things like this.

Reply:
> It is good practice to always validate user inputs. The fact that validation was done by 3rd party (azure while creation RG) is not right.

But azure is not a third-party it is the source of truth on this validation because if you tries to create a resource group that doesn't match the regex you mentioned Azure will not accept it. If we accept the user to provide us with an existing resource group, we do not care what name they have as long as Azure was happy with it. So I do not agree with your reasoning that we are incorrectly depending on _third party_ to do the validation.

> Hypothetical situation. In 6-month team members rotated, and somebody else comes in and implements "IfNotExistCreate" functionality. And you have a failure on your ends if they are not aware of things like this.

Yeah then the assumption of existing resource group does not hold and only then does it make sense to add the validation not right now.